clar2wasm/tests/wasm-generation/values.rs

```diff
@@ -58,23 +58,53 @@ proptest! {
 #[cfg(not(feature = "test-clarity-v1"))]
 mod clarity_v2_v3 {
     use clar2wasm::tools::TestEnvironment;
+    use clarity::vm::types::{BuffData, SequenceData};
     use clarity::vm::Value;
 
     use super::*;
-    use crate::runtime_config;
 
     proptest! {
-        #![proptest_config(runtime_config())]
-
         #[test]
-        fn value_serialized_and_deserialized(val in PropValue::any().prop_filter("Filter condition description", |val| {
+        fn serialize_deserialize_any_value(val in PropValue::any()) {
+            let snippet = format!("(to-consensus-buff? {val})");
+            const INPUT_TYPE_ERROR: &str = "could not determine the input type for the serialization function";
+
+            // Pre-check to discard test cases where `to-consensus-buff?` evaluation
+            // could not determine the type of the input parameter.
+            // For instance, `(to-consensus-buff? (list))` where a `NoType` should not be evaluated.
             let mut env = TestEnvironment::default();
-            env.evaluate(&format!("(to-consensus-buff? {val})")).is_ok()
-        })) {
+            let pre_check = env.evaluate(&snippet);
+            prop_assume!(match pre_check {
+                Ok(_) => true,
+                Err(ref e) if e.to_string().contains(INPUT_TYPE_ERROR) => false,
+                _ => true
+            });
+
+            // Serialize the PropValue to check it against
+            // the `to-consensus-buff?` implementation.
+            let serialized_value = Value::Sequence(SequenceData::Buffer(BuffData {
+                data: Value::from(val.clone()).serialize_to_vec().unwrap()
+            }));
```

Comment:
I have the same question as last time: how do we know the serialization will work here? You are unwrapping like a savage.

Reply:
Changed the serialization to outside the `data`.